docs/how_to/documenting_new_feature_toggles.rst

```diff
@@ -0,0 +1,94 @@
+.. _documenting_new_feature_toggles:
+
+=======================================
+How to: Documenting new feature toggles
+=======================================
+
+Feature toggles are boolean values that enable or disable the use of a certain feature. There are many of those across the Open edX codebase: in the form of Django settings, Waffle flags and switches, and  Configuration models, for example. It is important that these feature toggles be correctly annotated to address the following scenarios:
+
+- Generation of human-readable documentation of existing feature toggles for Open edX operators: instead of grepping a large codebase, operators (sysadmins and product owners, for instance) should be able to search an online document to learn how to use and enable features on their platforms.
+- Issue detection and resolution at runtime: feature toggles are exposed via an API on a running platform. This allows for automated discovery of feature toggles and, for instance, comparing the state of feature toggles between different environments at runtime. Annotations will allow operators to quickly understand the nature of the discrepancies, and to address issues more efficiently.
+
+Examples
+========
+
+Boilerplate template
+--------------------
+
+Copy-paste this boilerplate template to document a feature toggle::
+
+    # .. toggle_name: SOME_FEATURE_NAME
+    # .. toggle_implementation: WaffleFlag OR WaffleSwitch OR CourseWaffleFlag OR ExperimentWaffleFlag OR ConfigurationModel OR DjangoSetting
+    # .. toggle_default: True OR False
+    # .. toggle_description: Add here a detailed description of the consequences of enabling this feature toggle.
+    #   Note that all annotations can be spread over multiple lines by prefixing every line after the first by
+    #   at least two spaces, plus the leading space.
+    # .. toggle_warnings: Add here additional instructions that users should be aware of. For instance, dependency
+    #   on additional settings or feature toggles should be referenced here. If this field is not required, simply remove it.
+    # .. toggle_use_cases: temporary OR circuit_breaker OR vip OR opt_out OR opt_in OR open_edx
+    # .. toggle_creation_date: 2020-12-31
+    # .. toggle_target_removal_date: 2021-06-31 (this is required if toggle_use_cases is temporary. If not, simply remove it.)
```

Comment:
Let’s use 2020-01-01 and 2020-06-30. Quicker to see diff using the same year and June has 30 days.

Reply:
Correct! Actually, I'll use 2020-07-01 for a 6-month difference.